p2p/server_test.go

```diff
@@ -463,6 +463,108 @@ func TestServerSetupConn(t *testing.T) {
 	}
 }
 
+// TestServerPendingInboundRejection checks that duplicate inbound connections
+// are rejected when there's already a pending inbound connection from the same peer.
+func TestServerPendingInboundRejection(t *testing.T) {
+	trustedNode := newkey()
+	srv := &Server{
+		Config: Config{
+			PrivateKey:  newkey(),
+			MaxPeers:    10,
+			NoDial:      true,
+			NoDiscovery: true,
+			Logger:      testlog.Logger(t, log.LvlTrace),
+		},
+	}
+	if err := srv.Start(); err != nil {
+		t.Fatalf("could not start: %v", err)
+	}
+	defer srv.Stop()
+
+	newconn := func(id enode.ID) *conn {
+		fd, _ := net.Pipe()
+		tx := newTestTransport(&trustedNode.PublicKey, fd, nil)
+		node := enode.SignNull(new(enr.Record), id)
+		return &conn{fd: fd, transport: tx, flags: inboundConn, node: node, cont: make(chan error)}
+	}
+
+	// Create two connections from the same peer
+	peerID := randomID()
+	c1 := newconn(peerID)
+	c2 := newconn(peerID)
+
+	// First connection enters pendingInbound
+	err1 := srv.checkpoint(c1, srv.checkpointPostHandshake)
+	if err1 != nil {
+		t.Fatalf("first connection failed unexpectedly: %v", err1)
+	}
+
+	// Second connection should be rejected (duplicate pending inbound)
+	err2 := srv.checkpoint(c2, srv.checkpointPostHandshake)
+	if err2 != nil {
+		t.Fatalf("first connection passed unexpectedly: %v", err1)
+	}
+
+}
+
+// TestServerPendingInboundCleanup checks that pending inbound state is properly
+// cleaned up when a connection fails or completes.
+func TestServerPendingInboundCleanup(t *testing.T) {
```

Comment:
I'm not convinced that this is needed. Since the interesting behavior is local to dialScheduler and already covered by `TestDialSchedPendingInbound` in dial_test.go, I’d suggest dropping this test.